re/service.go

```diff
@@ -227,3 +235,80 @@ func (re *re) process(ctx context.Context, r Rule, msg *messaging.Message) error
 		return re.pubSub.Publish(ctx, m.Channel, m)
 	}
 }
+
+func (re *re) StartScheduler(ctx context.Context) error {
+	defer re.ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		case <-re.ticker.Tick():
+			startDateTime := time.Now()
+
+			pm := PageMeta{
+				Status:          EnabledStatus,
+				ScheduledBefore: &startDateTime,
+			}
+
+			page, err := re.repo.ListRules(ctx, pm)
+			if err != nil {
+				return err
+			}
+
+			for _, rule := range page.Rules {
+				if re.shouldRunRule(rule.Schedule, startDateTime) {
+					go func(r Rule) {
+						msg := &messaging.Message{
+							Channel: r.InputChannel,
+							Created: startDateTime.Unix(),
+						}
+						re.errors <- re.process(ctx, r, msg)
+					}(rule)
+				}
+			}
+		}
+	}
+}
+
+func (re *re) shouldRunRule(s Schedule, startTime time.Time) bool {
+	now := time.Now().Truncate(time.Minute)
+
+	// Don't run if the rule's start time is in the future
+	// This allows scheduling rules to start at a specific future time
+	if s.StartDateTime.After(now) {
+		return false
+	}
+
+	t := s.Time.Truncate(time.Minute)
+	if t.Equal(now) {
+		return true
+	}
+
+	switch s.Recurring {
+	case Daily:
+		if s.RecurringPeriod > 0 {
```

Comment:
Since each case includes this check, you can move it to before the switch: `if s.RecurringPeriod == 0 {return false}`.